pkg/ebpf/telemetry/errors_telemetry.go

```diff
@@ -256,3 +259,31 @@ func ebpfTelemetrySupported() (bool, error) {
 	}
 	return kversion >= kernel.VersionCode(4, 14, 0), nil
 }
+
+func elfBuildWithInstrumentation(bytecode io.ReaderAt) (bool, error) {
+	objFile, err := elf.NewFile(bytecode)
+	if err != nil {
+		return false, fmt.Errorf("failed to open elf file: %w", err)
+	}
+	defer objFile.Close()
+
+	const instrumentationSectionName = ".build.instrumentation"
+	sec := objFile.Section(instrumentationSectionName)
+	if sec == nil {
+		return false, nil
```

Comment:
if that's not an error, please state explicitly in the comment

Reply:
Done!